src/test/groovy/graphql/NullValueSupportTest.groovy

```diff
@@ -0,0 +1,249 @@
+package graphql
+
+import graphql.execution.NonNullableValueCoercedAsNullException
+import graphql.validation.ValidationError
+import graphql.validation.ValidationErrorType
+import spock.lang.Specification
+
+/*
+ * Taken from http://facebook.github.io/graphql/#sec-Input-Objects
+ *
+ *
+ 
+ Test Case   Original Value	        Variables	    Coerced Value
+ A          { a: "abc", b: 123 }	null	        { a: "abc", b: 123 }
+ B          { a: 123, b: "123" }	null	        { a: "123", b: 123 }
+ C          { a: "abc" }	        null	        Error: Missing required field b
+ D          { a: "abc", b: null }	null	        Error: b must be non‐null.
+ E          { a: null, b: 1 }	    null	        { a: null, b: 1 }
+ F          { b: $var }             { var: 123 }    { b: 123 }
+ G          { b: $var }             {}	            Error: Missing required field b.
+ H          { b: $var }             { var: null }	Error: b must be non‐null.
+ I          { a: $var, b: 1 }       { var: null }   { a: null, b: 1 }
+ J          { a: $var, b: 1 }       {}              { b: 1 }
+
+ */
+
+class NullValueSupportTest extends Specification {
+
+    def graphqlSpecExamples = '''
+        schema {
+            query : Query
+            mutation : Mutation
+        }
+        
+        type Query {
+            a : String
+            b: Int!
+        }
+        
+        type Mutation {
+            mutate(inputArg : ExampleInputObject) : Query
+        }
+            
+        input ExampleInputObject {
+             a: String
+             b: Int!
+        }
+            
+        '''
+
+    def "test graphql spec examples that output results"() {
```

Comment:
For input coercion from variables it must also testet that:
>  This unordered map should not contain any entries with names not defined by a field of this input object type, otherwise an error should be thrown.